langwatch/src/features/onboarding/components/sections/model-provider/ModelProviderConfigFields.tsx

```
@@ -0,0 +1,261 @@
+    });
+  }, []);
+
+  const handleOpenAiValidationClear = useCallback(() => {
+    setOpenAiValidationError(undefined);
+  }, []);
+
+  const validateOpenAi = useCallback(() => {
+    if (!isOpenAiProvider) return true;
+
+    const apiKey = state.customKeys.OPENAI_API_KEY?.trim() ?? "";
+    const baseUrl = state.customKeys.OPENAI_BASE_URL?.trim() ?? "";
+
+    // Both empty
+    if (!apiKey && !baseUrl) {
+      setOpenAiValidationError(
+        "Either API Key or Base URL must be provided",
+      );
+      return false;
+    }
+
+    // Base URL is set to default OpenAI URL, but no API key
+    if (baseUrl === OPENAI_DEFAULT_BASE_URL && !apiKey) {
+      setOpenAiValidationError(
+        "API Key is required when using the default OpenAI base URL",
+      );
+      return false;
+    }
+
+    handleOpenAiValidationClear();
+    return true;
+  }, [
+    isOpenAiProvider,
+    state.customKeys,
+    handleOpenAiValidationClear,
+  ]);
+
+  const handleSaveAndContinue = useCallback(() => {
+    if (!validateOpenAi()) {
+      return;
+    }
+
+    // Clear previous errors
+    setFieldErrors({});
+    handleOpenAiValidationClear();
+
+    // Validate keys according to schema before submitting
+    const providerDefinition = backendKey
+      ? modelProvidersRegistry[backendKey as keyof typeof modelProvidersRegistry]
+      : undefined;
+
+    if (providerDefinition?.keysSchema) {
+      const keysSchema = z.union([
+        providerDefinition.keysSchema,
+        z.object({ MANAGED: z.string() }),
+      ]);
+
+      const keysToValidate: Record<string, unknown> = { ...state.customKeys };
+      const result = keysSchema.safeParse(keysToValidate);
+
+      if (!result.success) {
+        // Parse the Zod error to get field-specific errors
+        const parsedErrors = parseZodFieldErrors(result.error as ZodErrorStructure);
+        setFieldErrors(parsedErrors);
+        return;
+      }
+    }
+
+    void actions.setEnabled(true).then(() => actions.submit());
```

We should try to handle rather than void and silence errors.

Maybe we need a simple helper:

```
.catch(easyCatch)
```